src/aioamazondevices/api.py

```diff
@@ -763,6 +783,128 @@ async def _response_to_json(self, raw_resp: ClientResponse) -> dict[str, Any]:
+                ):
+                    final_notifications.update(
+                        {
+                            schedule_device_serial: {
+                                **schedule_notification_list
+                                | {
+                                    schedule_type: AmazonSchedule(
+                                        type=schedule_type,
+                                        status=schedule_status,
+                                        label=schedule[label_desc],
+                                        next_occurrence=next_occurrence,
+                                    ),
+                                }
+                            }
+                        }
+                    )
+
+        return final_notifications
+
+    async def _parse_next_occurence(
+        self,
+        schedule: dict[str, Any],
+    ) -> datetime | None:
+        """Parse RFC5545 rule set for next iteration."""
+        # Local timezone
+        tzinfo = datetime.now().astimezone().tzinfo
+        # Current time
+        actual_time = datetime.now(tz=tzinfo)
+        # Reference start date
+        today_midnight = actual_time.replace(hour=0, minute=0, second=0, microsecond=0)
+        # Reference time (1 minute ago to avoid edge cases)
+        now_reference = actual_time - timedelta(minutes=1)
+
+        # Schedule data
+        original_date = schedule.get("originalDate")
+        original_time = schedule.get("originalTime")
+
+        recurring_rule: str | None = schedule.get("recurringRule")
+        if "rRuleData" in schedule:
+            recurring_rule = schedule["rRuleData"]["recurrenceRules"][0]
+
+        # Recurring events
+        if recurring_rule:
+            # Already in RFC5545 format
+            if "FREQ=" in recurring_rule:
+                return cast(
+                    "datetime",
+                    rrulestr(
+                        recurring_rule.removesuffix(";"), dtstart=today_midnight
+                    ).after(now_reference, True),
+                )
+
+            if recurring_rule not in RECURRING_PATTERNS:
+                _LOGGER.warning(
+                    "Unknown recurring rule: %s", scrub_fields(recurring_rule)
+                )
+                return None
+
+            # Adjust recurring rules for country specific weekend exceptions
+            recurring_pattern = RECURRING_PATTERNS.copy()
+            for group, countries in COUNTRY_GROUPS.items():
+                if self._country_code in countries:
+                    recurring_pattern |= WEEKEND_EXCEPTIONS[group]
+                    break
+
+            # Return the earliest next date
+            return cast(
+                "datetime",
+                rrulestr(
+                    recurring_pattern[recurring_rule], dtstart=today_midnight
+                ).after(now_reference, True),
+            )
+
+        # Single events
+        if schedule["type"] == NOTIFICATION_ALARM:
```

Comment:
```suggestion
        if schedule["type"] in NOTIFICATION_ALARM:
```

Reply:
Based on above change, this is not needed anymore